modules/packages/Python.chpl

```diff
@@ -2039,8 +2079,136 @@ module Python {
     }
   }
 
+  /*
+    Represents a Python tuple.  This provides a Chapel interface to Python
+    tuples, where the Python interpreter owns the tuple.
+  */
+  class PyTuple: Value {
+    @chpldoc.nodoc
+    proc init(in interpreter: borrowed Interpreter,
+              in obj: PyObjectPtr, isOwned: bool = true) {
+      super.init(interpreter, obj, isOwned=isOwned);
+    }
+
+    /*
+      Get the size of the tuple. Equivalent to calling ``len(obj)`` in Python.
+
+      :returns: The size of the tuple.
+    */
+    proc size: int throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var size = PyTuple_Size(this.getPyObject());
+      this.interpreter.checkException();
+      return size;
+    }
+
+    /*
+      Get an item from the tuple. Equivalent to calling ``obj[idx]`` in Python.
+
+      :arg T: The Chapel type of the item to return.
+      :arg idx: The index of the item to get.
+      :returns: The item at the given index.
+    */
+    pragma "docs only"
+    proc get(type T = owned Value, idx: int): T throws do
+      compilerError("docs only");
+
+    @chpldoc.nodoc
+    proc get(type T, idx: int): T throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var item = PyTuple_GetItem(this.getPyObject(), idx.safeCast(Py_ssize_t));
+      this.interpreter.checkException();
+      Py_INCREF(item);
+      return interpreter.fromPythonInner(T, item);
+    }
+
+    @chpldoc.nodoc
+    proc get(idx: int): owned Value throws do
+      return this.get(owned Value, idx);
+
+    /*
+      Get a slice of the tuple indicated by ``bounds``.  Equivalent to
+      calling ``obj[bounds.low:bounds.high+1]`` in Python.
+
+      .. note::
+
+         This method does not support strided ranges or ranges with an alignment
+         other than 0.
+
+      :arg T: the Chapel type of the slice to return.
+      :arg bounds: The full slice of the tuple to return
+      :returns: A slice of the tuple for the given bounds.
+    */
+    pragma "docs only"
+    proc get(type T = owned Value, bounds: range(?)): T throws do
+      compilerError("docs only");
+
+    @chpldoc.nodoc
+    proc get(type T, bounds: range(?)): T throws {
+      if (bounds.strides != strideKind.one) {
+        compilerError("cannot call `get()` on a Python tuple with a range with stride other than 1");
+      }
```

Comment:
is there any checking for a non-zero alignment?

Reply:
It seems like non-zero alignment can only occur [when the stride is not one](https://chapel-lang.org/docs/language/spec/ranges.html#align-operator), which was not something I understood before writing some tests